File: ninchatsdk/src/main/java/com/ninchat/sdk/adapters/NinchatQuestionnaireBaseAdapter.java
package com.ninchat.sdk.adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import com.ninchat.sdk.models.questionnaire.NinchatQuestionnaire;

import org.json.JSONArray;
import org.json.JSONObject;

public abstract class NinchatQuestionnaireBaseAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    private String TAG = NinchatQuestionnaireBaseAdapter.class.getSimpleName();
    protected NinchatQuestionnaire questionnaire;
    protected boolean isFormLikeQuestionnaire;

    public NinchatQuestionnaireBaseAdapter(NinchatQuestionnaire ninchatQuestionnaire, boolean isFormLikeQuestionnaire) {
        this.questionnaire = ninchatQuestionnaire;
        this.isFormLikeQuestionnaire = isFormLikeQuestionnaire;
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int position) {
        return null;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int position) {
    }

    public void addContent(JSONObject questionnaireList) {
        this.questionnaire.addQuestionnaire(questionnaireList);
    }

    public void updateContent(JSONArray questionnaireList) {
        this.questionnaire.updateQuestionnaireList(questionnaireList);
    }

    public JSONObject getLastElement() {
        return this.questionnaire.getLastElement();
    }

    public JSONObject getSecondLastElement() {
        return this.questionnaire.getSecondLastElement();
    }

    public void removeLast() {
        int position = getItemCount() - 1;
        this.questionnaire.removeQuestionnaireList(position);
        notifyItemRemoved(position);
    }

    public NinchatQuestionnaire getQuestionnaire() {
        return this.questionnaire;
    }

    public JSONArray getQuestionnaireList() {
        return this.questionnaire == null ? null : this.questionnaire.getQuestionnaireList();
    }

    @Override
    public int getItemCount() {
        return questionnaire == null ? 0 : questionnaire.size();
    }
}

File: ninchatsdk/src/main/java/com/ninchat/sdk/ninchatquestionnaire/ninchatbasequestionnaire/view/NinchatQuestionnaireBaseAdapter.kt
package com.ninchat.sdk.ninchatquestionnaire.ninchatbasequestionnaire.view

import androidx.recyclerview.widget.RecyclerView
import com.ninchat.sdk.models.questionnaire.NinchatQuestionnaire
import org.json.JSONArray
import org.json.JSONObject

abstract class NinchatQuestionnaireBaseAdapter(
        public var questionnaire: NinchatQuestionnaire?,
        protected var botDetails: Pair<String, String>,
        protected var isFormLikeQuestionnaire: Boolean,
) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun getItemViewType(position: Int): Int {
        return position
    }

    fun addContent(questionnaireList: JSONObject?) {
        questionnaire!!.addQuestionnaire(questionnaireList)
    }

    fun updateContent(questionnaireList: JSONArray?) {
        questionnaire!!.updateQuestionnaireList(questionnaireList)
    }

    val lastElement: JSONObject
        get() = questionnaire!!.lastElement
    val secondLastElement: JSONObject
        get() = questionnaire!!.secondLastElement

    fun removeLast() {
        val position = itemCount - 1
        questionnaire!!.removeQuestionnaireList(position)
        notifyItemRemoved(position)
    }

    val questionnaireList: JSONArray?
        get() = if (questionnaire == null) null else questionnaire!!.questionnaireList

    override fun getItemCount(): Int {
        return if (questionnaire == null) 0 else questionnaire!!.size()
    }
}